src/libsync/propagatorjobs.cpp

```diff
@@ -121,17 +126,38 @@ void PropagateLocalRemove::start()
         return;
     }
 
+    QString removeError;
     if (_item->isDirectory()) {
-        if (QDir(filename).exists() && !removeRecursively(QString())) {
-            done(SyncFileItem::NormalError, _error);
-            return;
+#ifdef Q_OS_UNIX
+        if (_moveToTrash) {
+            if (QDir(filename).exists() && !FileSystem::moveToTrash(filename, &removeError)) {
+                done(SyncFileItem::NormalError, removeError);
+                return;
+            }
+        } else
+#endif
+        {
+            if (QDir(filename).exists() && !removeRecursively(QString())) {
+                done(SyncFileItem::NormalError, _error);
+                return;
+            }
         }
     } else {
-        QString removeError;
-        if (FileSystem::fileExists(filename)
-            && !FileSystem::remove(filename, &removeError)) {
-            done(SyncFileItem::NormalError, removeError);
-            return;
+#ifdef Q_OS_UNIX
+        if (_moveToTrash) {
```

Comment:
Note: this code path is the same as for the directory one  (QDir(filename).exists()  can be replaced by FileSystem::fileExists.  So If you do the  `if (_moveToTrash) ` check  before the isDirectory check, you can greatly simplify the code